Support Laravel 9 LTS

LukeTowers opened the pull request:
See https://github.com/wintercms/winter/pull/148 for more information.

jaxwilko committed:
Added double quote escaping

jaxwilko committed:
Merge branch 'develop' into wip/config-writer-replacement

jaxwilko committed:
Updated ApplicationExceptions to SystemExceptions

jaxwilko committed:
Updated test to match the correct exception

jaxwilko committed:
Added support for setting env() args to correct types

jaxwilko committed:
Added config sort tests

jaxwilko committed:
Added sort() method for ConfigFile

jaxwilko committed:
Simplified internal sorting methods

jaxwilko committed:
Fixed test to be consitant across php versions (RFC: stable_sorting)

jaxwilko committed:
Switched line ending to unix style when creating empty file

jaxwilko committed:
Switched line ending for rendered output

jaxwilko committed:
Switched heredoc test samples for quoted strings

jaxwilko committed:
Removed incorrect newline at end of expected results

jaxwilko committed:
Reverted to heredoc and removed carriage returns from expected samples

jaxwilko committed:
Removed usage of PHP_EOL in favour of lf

jaxwilko committed:
Added class to represent const expressions

jaxwilko committed:
Fixed incorrect comment

jaxwilko committed:
Added tests for setting arrays and const values

jaxwilko committed:
Fixed incorrect method description comment

jaxwilko committed:
Added support for setting an array and const value

jaxwilko committed:
Simplified key insertion logic

jaxwilko committed:
Added test for numeric array keys

jaxwilko committed:
Fixed spelling mistake

jaxwilko committed:
Swapped out redundent if statement for assigning result of expression

LukeTowers committed:
Merge branch 'develop' into wip/1.2

LukeTowers committed:
Update dependencies

LukeTowers committed:
Improve support PHP 8.1 & align return types with Laravel base classes

See https://github.com/laravel/framework/pull/40471

LukeTowers committed:
Match Laravel 9 langPath() interface

See https://github.com/laravel/framework/commit/e6c8aaea886d35cc55bd3469f1a95ad56d53e474

LukeTowers committed:
Compatibility with Laravel 9 Cache Repository changes

See https://github.com/laravel/framework/pull/40249

mjauvin commented on src/Foundation/Application.php:
```diff
@@ -70,7 +70,7 @@ public function publicPath()
-    public function langPath()
+    public function langPath($path = '')
     {
         return PathResolver::join($this->basePath, '/lang');
```
```suggestion
        return PathResolver::join($this->basePath, '/lang' . (!empty($path) ? "/$path" : ''));
```

LukeTowers committed:
Update src/Foundation/Application.php

Co-authored-by: Marc Jauvin <marc.jauvin@gmail.com>

LukeTowers committed:
Remove unnecessary Str facade

Was entirely unnecessary from the very start as all Str helper methods are static methods. Not sure why it was originally included in https://github.com/wintercms/storm/commit/89c73a6fff53f8744ade104b9f93794c248a46dc.

LukeTowers committed:
Switch back to stable version of Tinker

LukeTowers committed:
Ensure any return code from the parent laravel command is returned

LukeTowers committed:
Update testing dependencies to match Laravel 9

LukeTowers committed:
Add autocompletion support to the create:command stub

Refs: wintercms/winter#450

LukeTowers committed:
Remove Laravel migration commands

They never worked with Winter so they have been removed. Aliases as relevant will be setup to point to Winter's equivalent commands

LukeTowers committed:
Don't load module route files when routes are cached

LukeTowers committed:
Add commands to review for inclusion in Winter

LukeTowers committed:
Fix support for route:list & route:cache commands

Moves module route registration into the boot method where it's supposed to go.

> You should never attempt to register any event listeners, routes, or any other piece of functionality within the register method.
- https://laravel.com/docs/9.x/providers#writing-service-providers

This previous logic (where modules setup a deferred registration of their routes by binding to the App::before() event (injecting into the router immediately prior to dispatching requests; and this binding itself happens in the register() method of the serviceprovider) was originally added in order to support plugins being able to override the greedy final route provided by the CMS module. See https://github.com/wintercms/winter/commit/04bbad320fb033e2e0ebde001b58a469ff6164ec for more details.

This logic overcomplicated things however and caused issues with Laravel tools that expected to have access to all routes regardless of whether an actual request was about to be dispatched or if the application was actually being run in the CLI. In October 2.0 the approach to add another method and layer of hackiness in the form of registerLateRoutes() (which manually dispatches a blank request in order to trick the app::before into firing and registering module routes); however this solution is overcomplicated and too brittle in my personal opinion so instead I have opted to register module routes in a more Laravel-like way, where they are registered when the ServiceProvider is booted.

At some point in the future we may want to get rid of the App::before() / after() (and corresponding router events as well) altogether since there are usually better ways to achieve similar results without using them and completely non-obviously those events do not run in the CLI at all since there is no request being dispatched.

LukeTowers committed:
Bump Twig to v3.0

Note: The {% spaceless %} tag has been removed in 3.0, we may add a polyfill back for it in Winter though.

LukeTowers committed:
Merge 1.2 into config-writer-replacement

LukeTowers committed:
Move config writer into Parse

LukeTowers committed:
Move ArrayFile tests out of config and into parse

LukeTowers committed:
Various cleanup and code review

LukeTowers committed:
Fix test name mismatch

LukeTowers committed:
PHPConst -> PHPConstant

LukeTowers committed:
FileInterface -> DataFileInterface, read() -> open()

LukeTowers committed:
Require PHP Codesniffer > 3.2

LukeTowers committed:
Switch back to Laravel's default logic for setting application keys

Difference that justifies us continuing to override the command is that we will automatically create the .env file if it doesn't exist.

LukeTowers committed:
const() -> constant()

LukeTowers committed:
Improvements to the EnvFile parser

LukeTowers committed:
Use Str facade instead of helper functions

LukeTowers committed:
Cleanup docblocks

LukeTowers committed:
Fix tests, add test case for $throwIfMissing

LukeTowers committed:
Refactor ConfigWriter to use ArrayFile parser internally

LukeTowers committed:
Fix Mail::pretend() usage

LukeTowers committed:
Switch to Symfony\Yaml v6

Still some work needed to support legacy YAML uses

LukeTowers committed:
Temporary workaround for version.yaml parsing

mjauvin committed:
cannot use facades with tests

LukeTowers committed:
Replace embedded Assetic v1 with Assetic v3

Also replaces linkorb/jsmin-php with wikimedia/minify and reordered main dependencies to be alphabetical

LukeTowers committed:
Merge branch 'wip/1.2' of github.com:wintercms/storm into wip/1.2

LukeTowers committed:
Fix Assetic -> October\Rain aliasing

bennothommo committed:
Allow anonymous migration classes

Refs: https://github.com/wintercms/winter/pull/148

bennothommo committed:
Define return var for singleton instance.

Allows IDEs to autocomplete parameters and functions in a singleton instance.

LukeTowers committed:
Fix YAML processor tests

bennothommo committed:
Remove scaffold commands from Storm (#70)

See wintercms/winter#471 for more information.

LukeTowers committed:
Update comments

mjauvin committed:
Fix closure listener with parameters as reference (#69)

Previously listeners would be wrapped in a `SerializedClosure` before being passed to Laravel which would then wrap them in a regular `Closure`. This change ensures that we first wait for Laravel to wrap the listeners in the required custom logic before we then wrap the prepared `Closure` in a `SerializedClosure`.

Fixes issue with event listeners that rely on arguments being passed as reference: "Argument #1 ($key) must be passed by reference, value given"

LukeTowers committed:
cleanup docblock

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Update Composer deps, small tweaks

bennothommo committed:
Add class comment

bennothommo committed:
Fix Stan issues in Translator

LukeTowers committed:
Add Command base class

Adds the Winter\Storm\Console\Command base class that provides some utilities to make developing CLI commands easier.

Commands can extend this class and then provide completion / suggestions for arguments / options by adding suggest{$name}[Values|Options] methods for each of the arguments / options that it wants to provide suggestions for.

bennothommo committed:
Fix Validation and Support PHPStan issues

bennothommo committed:
Fix Router PHPStan issues

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

LukeTowers committed:
Provide example CLI argument value suggestion method implementation

bennothommo committed:
Add Event facade methods

bennothommo committed:
Fix Parse PHPStan issues

bennothommo committed:
Fix Network PHPStan issues

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Fix return type of fired events

bennothommo committed:
Fix minor issue in MailManager

bennothommo committed:
WIP Mailer fixes

bennothommo committed:
Fix HTML PHPStand issues

bennothommo committed:
Fix unit tests - Mail ones are not passing now, but pass on main branch

LukeTowers committed:
Code quality

LukeTowers committed:
Add support for wordwrapping the console alert() output helper

bennothommo committed:
Implement changes from Luke's review

bennothommo committed:
Finalise Mail PHPStan fixes

LukeTowers committed:
Add ConfirmsWithInput CLI trait

Provides the confirmWithInput($message, $requiredInput) method to CLI commands to show an alert with the provided message and a prompt to input the required input to act as confirmation that the command should be run.

LukeTowers committed:
Include Winter CMS in the app version

LukeTowers committed:
Remove dependency on App facade

bennothommo committed:
Merge remote-tracking branch 'origin/wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Fix Command PHPStan issues

LukeTowers committed:
Add # as a filesystem path symbol to refer to themes_path()

LukeTowers committed:
Docblocks cleanup

bennothommo committed:
Merge remote-tracking branch 'origin/wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Tweak extendable trait

bennothommo committed:
WIP fixes of Database PHPStan issues

bennothommo committed:
Defensive coding tweak

LukeTowers committed:
Improve Halcyon SectionParser logic

In addition to adding method signature typehints and general code tidying; this commit improves the ability of the SectionParser to handle extraneous section separators in appearing in settings, code, & markup content.

With these changes; if extra separators are found in the markup section, instead of ignoring the extra sections they will be considered part of the markup section. Additionally, if extra sections are found in the markup section and there was previously only 1 or 2 sections total (markup only or settings and markup) the SectionParser::render() method will force there to be 3 sections in order to guarantee additional separators found in the markup section stay in the markup section.

Another change to the SectionParser::render() method is that it will now throw an InvalidArgumentException if section separators are found in either the settings section or the code section as it is not currently possible to avoid accidentally parsing separators in those locations as if they were valid and intentional separators.

This is somewhat related to https://github.com/octobercms/library/commit/c393c5ce9ca2c5acc3ed6c9bb0dab5ffd61965fe although instead of just dropping all possible sections within markup and keeping the last one - this change implements more robust support for section separators appearing in places where they shouldn't.

LukeTowers committed:
Code quality and windows test fix

LukeTowers committed:
Merge branch 'develop' into wip/1.2

LukeTowers committed:
Remove the Seed & StorageLink Laravel commands

These commands aren't used in Winter and have alternatives (winter:up & winter:mirror).

bennothommo committed:
Merge remote-tracking branch 'origin/wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Adjust Command logic path for input type parsing

bennothommo committed:
Fix some issues picked up by PHPStan when global scan is used

LukeTowers committed:
Fixes to the Mailer

Refs: https://github.com/laravel/framework/commit/63ca843643e86fb69efc901051ae079c89a7fd09

jaxwilko committed:
Added config test fixtures

jaxwilko committed:
Added support for leading imports & expressions before a return stmt

jaxwilko committed:
Manual merge wip/1.2 into wip/config-writer-replacement

jaxwilko committed:
Moved fixtures into arrayfile dir

jaxwilko committed:
Added include test to ensure parens on correct include stmts

jaxwilko committed:
Added code to check include position in ast and append parens correctly

jaxwilko committed:
Simplified code by always apending parens on include/require stmts

bennothommo committed:
Drop polyfills

No longer needed with L9 branch as the minimum PHP version is 8.0

LukeTowers committed:
remove polyfills require statement

LukeTowers committed:
Merge branch 'wip/1.2' into wip/config-writer-replacement

LukeTowers committed:
Trim useless whitespace

@jaxwilko's favourite piece of the whole ArrayFile parser ;)

jaxwilko committed:
Added fixes for whitespace issues

mjauvin committed:
Allow array of email addresses without names (#75)

LukeTowers committed:
Typehint the app property for moduleserviceproviders

LukeTowers committed:
Improve DX for dynamically adding relationships

This improves the developer experience for dynamically adding relationships to an external model when extending the model. Previously the developer would have to set the relation property directly themselves by merging their additions with the existing definitions and usually they wouldn't include any error checking logic.

LukeTowers committed:
Wait for the dispatcher to be set before booting model events

This fixes an issue that occurs when a model class is instantiated before the DatabaseServiceProvider can be booted (thus calling Model::setEventDispatcher()). Previously Winter would proceed with booting the nicer events, and then helpfully set a flag the events for the current model class had already been booted; however this would cause any and all "nice" model events to fail since they were never actually registered with an event dispatcher since Laravel silently discarded the required listeners when the $dispatcher property wasn't set.

jaxwilko committed:
Bound lexer to ArrayFile instance

jaxwilko committed:
Added support for lexer token parsing and additional comment handling

jaxwilko committed:
Added non-attribute comment handling test

jaxwilko committed:
Added style fixes

jaxwilko committed:
More styling fixes in test fixture

jaxwilko committed:
Added fix to ConfigWriter to support passing lexer to ArrayFile instance

jaxwilko committed:
Added \r trim to comment test

bennothommo committed:
Update src/Foundation/Console/KeyGenerateCommand.php

LukeTowers committed:
KeyGenerateCommand cleanup

bennothommo committed:
Code dusting and explicit typing from review

jaxwilko committed:
Added subitem comment test

jaxwilko committed:
Added fix for subitems being picked up in lexer tokens

jaxwilko committed:
Merge branch 'wip/config-writer-replacement' of github.com:wintercms/storm into wip/config-writer-replacement

jaxwilko committed:
Added fix for token parsing running into comma tokens and breaking

LukeTowers committed:
Merge pull request #40 from wintercms/wip/config-writer-replacement

[Enhancement] Add ArrayFile & EnvFile Parsing, replace ConfigWriter internal logic

LukeTowers committed:
Style fix

bennothommo committed:
Check for a space / end before surrounding keys with quotes

LukeTowers committed:
Revert "Check for a space / end before surrounding keys with quotes"

This reverts commit cd4b82d3485c5cbfb79016daca6d684502e7e150.

LukeTowers committed:
Only apply YAML preprocessing if the initial parse attempt fails

LukeTowers committed:
Fix tests

LukeTowers committed:
Align GeneratorCommand with Laravel (#76)

Improves the alignment of the base GeneratorCommand with the Laravel implementation. Related: https://github.com/wintercms/winter/pull/486

LukeTowers committed:
Fix scaffolding path overwriting logic

mjauvin committed:
Fire mailer.beforeRegister event in mailer() method (#79)

Related: https://github.com/wintercms/winter/pull/522

This fix is needed because if the event is fired in resolve() method, the $name variable has already been defined (from a call to getDefaultDriver() in the base mailer() method)

ref. https://github.com/laravel/framework/blob/9.x/src/Illuminate/Mail/MailManager.php#L70

Note: if the $name argument is provided to the mailer() method, the problem does not occur because the config is forced to that driver name.

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

mjauvin committed:
remove non-existent argument to Ini::parse() method (#82)

bennothommo committed:
Merge branch 'develop' into wip/1.2

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Fix tests

bennothommo committed:
Synchronise docblocks

bennothommo committed:
Drop unnecessary overwrite in hasOne relation.

"getResults" overwrite is already handled correctly in Laravel.

bennothommo committed:
Simplify "key:generate" command overwrite

bennothommo committed:
Tweak checks for URL generator values

bennothommo committed:
Refactor database relations.

- Move all traits into a concerns folder, to keep only supported relations in the base relation folder. Stubs for previously available traits have been kept for BC, but are deprecated.
- Abstract BelongsToMany functionality into a trait to share with this class as well as MorphsToMany.
- Change MorphToMany to extend the base MorphToMany class from Laravel, to maintain parity and covariance.
- Fix multiple PHPStan issues

bennothommo committed:
Make Argon extend Illuminate's Carbon wrapper

bennothommo committed:
Numerous fixes to docblocks and signatures

bennothommo committed:
Exclude some errors that are not helping

bennothommo committed:
Final pass of code analysis tweaks

bennothommo committed:
Fix sortable constant reference and test

bennothommo committed:
Convert private methods to protected in Section Parser

bennothommo committed:
Change return type in Argon

bennothommo committed:
Synchronise Pivot model with Laravel 9

bennothommo committed:
Add MorphPivot support

bennothommo committed:
Update src/Database/Attach/File.php

bennothommo committed:
Update src/Console/Command.php

Co-authored-by: Luke Towers <github@luketowers.ca>

bennothommo committed:
Update src/Database/MorphPivot.php

Co-authored-by: Luke Towers <github@luketowers.ca>

bennothommo committed:
Add type hints to all File attachment methods

bennothommo committed:
Add type hints

bennothommo committed:
Make the Pivot use the AsPivot trait, add parent accessor

bennothommo committed:
Update src/Database/Behaviors/Purgeable.php

Co-authored-by: Luke Towers <github@luketowers.ca>

bennothommo committed:
Update src/Database/Behaviors/Sortable.php

Co-authored-by: Luke Towers <github@luketowers.ca>

LukeTowers committed:
Update src/Filesystem/Filesystem.php

LukeTowers committed:
Fix support for Pivot models in 1.2

Credit to @bennothommo in https://github.com/wintercms/storm/commit/a78075898ff70e6114c3339bfd690b4c0a99b085. Replaces part of https://github.com/wintercms/storm/pull/65

LukeTowers committed:
Added ProcessesQuery console helper trait to make processing large numbers of records easier.

LukeTowers committed:
Ported laravel cache provider without implementing deferable (#84)

Required for wintercms/winter#501. See also wintercms/winter#493
Also removed upsert tests ref: 04230d5ec44cf0b5f354e918f2c5a6079936339f

jaxwilko committed:
Removed deferable from redis provider (#85)

See https://github.com/wintercms/winter/pull/501

bennothommo committed:
Add Array Source trait (#87)

This commit introduces the ability to create an "array source" model which provides arbitrary records via an array. These models will act, for all intents and purposes, as Eloquent models, but do not require a database.

The implementation of this feature is inspired by the Sushi library by Caleb Porzio (https://github.com/calebporzio/sushi) but modified to work in Winter's architecture.

This commit also introduces a change to the testing suite, to cater for the facades used in this trait, and now creates a dummy application when running tests, which resolves a long-standing issue where facades in the test cases did not work.

bennothommo committed:
Tweaks to array source trait

As per comments for https://github.com/wintercms/storm/commit/221dcb2c89197b7c09036c5b796131bba8c1d76e

Thanks to @LukeTowers and @mjauvin

jaxwilko committed:
Added support for setting app tempPath via config (#89)

Especially useful for Laravel Vapor where the only local writable directory is `/tmp`.

bennothommo committed:
Restores the "addVisible" / "addHidden" methods removed in Laravel 7

It is recommended to use "makeVisible" / "makeHidden" in new code going forward.

Fixes https://github.com/wintercms/winter/issues/567

bennothommo committed:
Load deferred providers before provider registration (#86)

Reverts dff4e6e2f150c059fd1934e4cac3dcbc99af638e
Partially reverts 6d950462bf8aaf8a3d5189030797b6b39b2ea7d6

Implements laravel/framework#38724 into the Winter codebase, extending the base Laravel ProviderRepository class and overriding the load method.

This will override the base provider loading functionality to record deferred providers before registering any eager loaded providers. This allows eager loaded providers to use deferred provider functionality (such as the Cache provider). In this method, this will technically make those providers eager loaded as well, but these providers will be deferred if no other provider uses them in registration.

LukeTowers committed:
Bump minimum Laravel version to 9.1

See https://github.com/laravel/framework/commit/63ca843643e86fb69efc901051ae079c89a7fd09

LukeTowers committed:
Merge branch 'develop' into wip/1.2

* develop:
  Avoid issues coming from double encoded URLs

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Remove relation traits that have been moved to concerns

bennothommo committed:
Update all facade use cases to use full facade class

bennothommo committed:
Add options docs for Zip class, fix BC break

bennothommo committed:
Re-add default value

bennothommo committed:
Remove overwritten definition for resolver

LukeTowers committed:
Merge branch 'develop' into wip/1.2

* develop:
  Added additional test case for UrlGenerator class (#90)
  Improve port validation when building URLs

bennothommo committed:
Update test case for choice / transChoice

bennothommo committed:
Bring Translator closer to base Laravel functionality.

Refactors the "get" override to use the base "get" method for the most part, removes the "choice" method override and adds the "set" method in Translator.

Changes to the FileLoader will allow our original fix for the sublocales in the "choice" method to work even with the base Laravel functionality.

The FileLoader will now attempt to find locales in the format "xx_XX" as per the original Laravel functionality and then also attempt "xx-xx" as per the traditional usage in Winter.

bennothommo committed:
Don't try to double-check lang files unnecessarily

bennothommo committed:
Load JSON files before setting translation value in namespace

bennothommo committed:
Add Laravel test cases to ensure base functionality works

bennothommo committed:
Remove "translator.beforeResolve" event for performance reasons

bennothommo committed:
Remove test for "translator.beforeResolve" event

bennothommo committed:
Add ability to change an entire group through "set" method

bennothommo committed:
Ensure group override is not applied when locale is changed back

LukeTowers committed:
Merge pull request #91 from wintercms/wip/1.2-translation-improvements

Bring Translator closer to base Laravel functionality, add "set" method, remove translator.beforeResolve event.

Fixes wintercms/winter#480

bennothommo committed:
Merge branch 'wip/1.2' into wip/1.2-code-analysis

bennothommo committed:
Fix moved class names in aliases

bennothommo committed:
Add MorphPivot test cases

bennothommo committed:
Fix constructor for MorphToPivot

bennothommo committed:
Revert hour value change

bennothommo committed:
Correct docblock return

bennothommo committed:
Match PHPCS config from main repo

bennothommo committed:
Use lang path in Translator registration

bennothommo committed:
Fix possible BC break, cleanup for PHPStan

bennothommo committed:
Code tidying

bennothommo committed:
Try out @phpstan-property definitions to fix remaining issues

bennothommo committed:
Docs and commenting changes to fix a whole slew of issues

bennothommo committed:
Generate baseline

bennothommo committed:
Allow pivot model to be provided to MorphTo relations

bennothommo committed:
Fix test, allow namespaced files in tests folder

bennothommo committed:
Move tests namespace into autoload-dev

bennothommo committed:
Add automated PHPStan test, tweak PHPUnit test

bennothommo committed:
Replace missing matrix vars

bennothommo committed:
Fix Markdown parser class name storage and use

bennothommo committed:
Remove use of Eloquent facade in down migrations

bennothommo committed:
Convert all type hints to PHPDocs to maintain BC

bennothommo committed:
Small tweak

bennothommo committed:
Ensure morph to supports deferred records and constraints

bennothommo committed:
Remove fixed errors from baseline

bennothommo committed:
Allow CSRF token and session key to be null

bennothommo committed:
Fix return type

bennothommo committed:
Ignore lines in PHPStan - it can't seem to work out the relations here

bennothommo committed:
Update src/Database/Relations/Concerns/DeferOneOrMany.php

Co-authored-by: Luke Towers <github@luketowers.ca>

bennothommo committed:
Update src/Database/Relations/Concerns/DeferOneOrMany.php

Co-authored-by: Luke Towers <github@luketowers.ca>

bennothommo committed:
Remove return type missed in last pass

bennothommo committed:
Change to Str call

bennothommo committed:
Merge pull request #64 from wintercms/wip/1.2-code-analysis

Code analysis for Laravel 9 branch

jaxwilko committed:
Added a __toString to resolve Twig issue (#93)

Temporary fix while waiting for https://github.com/twigphp/Twig/pull/3719 to be merged.

mjauvin committed:
Restore Laravel 6 behavior to public visibility for local driver (#92)

Fixes https://github.com/wintercms/winter/issues/503

LukeTowers committed:
Only apply default visibility of public to the default local disk

See https://github.com/wintercms/winter/commit/f05aba73877a06d23aed4ab7e16c7a27bd011cb2

teranode committed:
Improve support for PHP 8.1 (#81)

See https://github.com/wintercms/winter/pull/524

mjauvin committed:
add unit test with doubly encoded url in URL path

mjauvin committed:
Revert "add unit test with doubly encoded url in URL path"

This reverts commit f3b0aa99f41654f5ab3fdb311eb14b831f2c2356.

bennothommo committed:
Use path resolver to determine base directory for array data cache

mjauvin committed:
Fix for placeholder default block (#96)

Restores previous default value for BlockBuilder placeholder/get methods (changed in https://github.com/wintercms/storm/commit/1a46e9e9ecea3e71c15cbe8174708724466bcc15 which caused issues as the `placeholderFunction` method in the CMS module was checking for null, not an empty string. 

Technically speaking a block that doesn't exist with no default value provided should return null, so the original default parameter values were correct.

LukeTowers commented on src/Auth/Manager.php:
```diff
@@ -1,8 +1,8 @@
-use Cookie;
-use Session;
-use Request;
+use \Illuminate\Support\Facades\Cookie;
+use \Illuminate\Support\Facades\Session;
+use \Illuminate\Support\Facades\Request;
```
```suggestion
use Illuminate\Support\Facades\Cookie;
use Illuminate\Support\Facades\Session;
use Illuminate\Support\Facades\Request;
```

LukeTowers committed:
Update src/Auth/Manager.php

LukeTowers commented on src/Halcyon/Datasource/ResolverInterface.php:
```diff
@@ -1,28 +1,48 @@
+     * Get a datasource instance by name.
      *
-     * @param  string  $name
+     * @param string $name The name of the datasource to retrieve.
      * @return \Winter\Storm\Halcyon\Datasource\DatasourceInterface
+     * @throws \Winter\Storm\Halcyon\Exception\MissingDatasourceException If a datasource with the given name does not exist.
```
```suggestion
     * @throws \Winter\Storm\Halcyon\Exception\MissingDatasourceException If a datasource with the given name does not exist.
```

LukeTowers committed:
Update src/Halcyon/Datasource/ResolverInterface.php

LukeTowers committed:
Code review

mjauvin committed:
Only decode query arguments before parsing the URL (#95)

Fixes issue introduced in 32df10c that caused double encoded URLs present in a URL's path (double encoded to survive Laravel's router that decodes path segments before routing) to be decoded during UrlGenerator actions which would break them as the output would no longer be double encoded as was specifically provided to the input. 

The above commit broke the ImageResizer encoded URLs in wintercms/winter

LukeTowers commented:
**NOTE** v1.1.9 will need to be reviewed and made live before this PR is merged into develop.

LukeTowers committed:
Merge branch 'develop' into wip/1.2

* develop:
  Apply fix from 1.0
  Add test case for trimStringAttributes flag
  Make string attribute value auto-trimming optional (#97)

# Conflicts:
#	src/Database/Attach/File.php
